Source/RunActivity/Viewer3D/Weather.cs

```diff
@@ -914,6 +923,94 @@ public AutomaticWeather(Viewer viewer)
+                    CheckValue(ref thisPrecipitation.OvercastVisibility, false, 1000, 60000, acttime, "Overcast Visibility");
+
+                    // precipitation
+                    CheckValue(ref thisPrecipitation.PrecipitationDensity, true, 0, 1, acttime, "Precipitation Density");
+                    CheckValue(ref thisPrecipitation.PrecipitationVariation, true, 0, 1, acttime, "Precipitation Variation");
+                    CheckValue(ref thisPrecipitation.PrecipitationRateOfChange, true, 0, 1, acttime, "Precipitation Rate Of Change");
+                    CheckValue(ref thisPrecipitation.PrecipitationProbability, true, 0, 100, acttime, "Precipitation Probability");
+                    CheckValue(ref thisPrecipitation.PrecipitationSpread, false, 1, 1000, acttime, "Precipitation Spread");
+                    CheckValue(ref thisPrecipitation.PrecipitationVisibilityAtMinDensity, false, 100, thisPrecipitation.OvercastVisibility, acttime, "Precipitation Visibility At Min Density");
+                    CheckValue(ref thisPrecipitation.PrecipitationVisibilityAtMaxDensity, false, 100, thisPrecipitation.PrecipitationVisibilityAtMinDensity, acttime, "Precipitation Visibility At Max Density");
+
+                    // build up
+                    CheckValue(ref thisPrecipitation.OvercastPrecipitationStart, true, thisPrecipitation.Overcast, 100, acttime, "Overcast Precipitation Start");
+                    CheckValue(ref thisPrecipitation.OvercastBuildUp, true, 0, 1, acttime, "Overcast Build Up");
+                    CheckValue(ref thisPrecipitation.PrecipitationStartPhase, false, 30, 240, acttime, "Precipitation Start Phase");
+
+                    // dispersion
+                    CheckValue(ref thisPrecipitation.OvercastDispersion, true, 0, 1, acttime, "Overcast Dispersion");
+                    CheckValue(ref thisPrecipitation.PrecipitationEndPhase, false, 30, 360, acttime, "Precipitation End Phase");
+                }
+
+                else if (weatherSet.GetType() == typeof(WeatherSettingFog))
+                {
+                    WeatherSettingFog thisFog = weatherSet as WeatherSettingFog;
+                    CheckValue(ref thisFog.FogOvercast, true, 0, 100, acttime, "Fog Overcast");
+                    CheckValue(ref thisFog.FogSetTime, false, 300, 3600, acttime, "Fog Set Time");
+                    CheckValue(ref thisFog.FogLiftTime, false, 300, 3600, acttime, "Fog Lift Time");
+                    CheckValue(ref thisFog.FogVisibility, false, 10, 20000, acttime, "Fog Visibility");
+                }
+            }
+        }
+
+        // check value, set random value if allowed and value not set
+        void CheckValue(ref float setValue, bool randomize, float minValue, float maxValue, TimeSpan acttime, string description)
+        {
+            // overcast
+            if (setValue < 0 && randomize)
```

Comment:
Since you're checking for negative here, the ternary that explicitly sets them to -1 when loading is unnecessary.